src/Common/src/System/PasteArguments.cs

```diff
@@ -29,18 +30,35 @@ public static string Paste(IEnumerable<string> arguments, bool pasteFirstArgumen
                     //   - Parsing ends at first whitespace outside quoted region.
                     //   - No way to get a literal quote past the parser.
 
+                    bool hasQuote = false;
                     bool hasWhitespace = false;
                     foreach (char c in argument)
                     {
                         if (c == Quote)
                         {
-                            throw new ApplicationException("The argv[0] argument cannot include a double quote.");
+                            hasQuote = true;
                         }
                         if (char.IsWhiteSpace(c))
                         {
                             hasWhitespace = true;
                         }
                     }
+
+                    if (hasQuote)
+                    {
+                        // If the argv[0] contains a double-quote, we're in a pickle because there is no string that would generate such an argv[0] 
+                        // under our parsing rules.
+                        //
+                        // Right now, the only user of this method that asks for arv[0] parsing is Environment.CommandLine. Since it's passing us an arg array that came from
+                        // framework itself as a result of tokenizing a command line under these same rules, this situation should never come up. But if it ever does,
+                        // throwing an exception for this seems extreme given that it's likely that the command line is just being retrieved for logging purposes
```

Comment:
It's hard to predict the future. (Actually, I think the UWP case already takes arguments externally rather than segmenting so "never" is too strong.) I just know that if this contingency ever comes up, I would rather that this api return something I can read in a log rather than throw preemptively on the unproven assumption that the app is going to try to retokenize it.